### PR TITLE
5826_log_warnings

### PR DESCRIPTION
**Pull request by LauraGTyler**
This removes some twig 3 and php8 warnings and depreciation messages..

### CODE REVIEW

File: lib/Basket.php [hunk 1]
```diff
@@ -244,13 +244,15 @@ public function setAddressStatus( $isReady, $countryCode, $addressString )
      */
     public function getEUVatCountries()
     {
-        $ret = [ ];
-        foreach ($this->state->vatRates[ 'rates' ] as $cc => $info) {
+	$ret = [ ];
+        if(!empty($this->state->vatRates[ 'rates' ])){
```
Comment (patternseek) [inline lib/Basket.php]: $this->state->vatRates being empty would be a bug so we shouldn't be working around it here

Comment (LauraGTyler) [reply lib/Basket.php]: So this is addressing the following warnings on going to the dev server with a checked out patternseek:

Jun 27 13:22:35 centralindex wolf-app[224389]: PHP Warning:  Undefined array key "rates" in /home/webuser/patternseek/ecommerce/lib/Basket.php on line 248
Jun 27 13:22:35 centralindex wolf-app[224389]: default.WARNING: E_WARNING: foreach() argument must be of type array|object, null given {"code":2,"message":"foreach() argument must be of type array|object, null given","file":"/home/webuser/patternseek/ecommerce/lib/Basket.php","line":248} []

Comment (LauraGTyler) [reply lib/Basket.php]: So this is triggered by an error dev mode whereby the vat rates are not coming down.
http://bugz.getinstance.com/show_bug.cgi?id=5914
When this bug is fixed I'll checkin a change that removes this fix.. so change here pending..

Comment (LauraGTyler) [reply lib/Basket.php]: change checked in below

**Review by patternseek — APPROVED** [review]
CODE: Yes

**Pull request merged**